http/src/main/java/io/micronaut/http/cookie/ServerCookieDecoder.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2017-2024 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.http.cookie;
+
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.core.io.service.ServiceDefinition;
+import io.micronaut.core.io.service.SoftServiceLoader;
+
+import java.util.List;
+
+/**
+ * Constructs cookies from an HTTP header string.
+ * @author Sergio del Amo
+ * @since 4.3.0
+ */
+@FunctionalInterface
+public interface ServerCookieDecoder {
+    /**
+     * The default {@link ServerCookieDecoder} instance.
+     */
+    ServerCookieDecoder INSTANCE = SoftServiceLoader
+            .load(ServerCookieDecoder.class)
+            .firstOr("io.micronaut.http.cookie.DefaultServerCookieDecoder", ServerCookieDecoder.class.getClassLoader())
+            .map(ServiceDefinition::load)
+            .orElse(null);
```

Comment:
we should probably initialise these at build time for Graal

Reply:
the `nativeTest` seems to pass. 

Do you want me to create a `native-image.properties` in the `http` module with content: 

```
Args = --initialize-at-build-time=io.micronaut.http.cookie
```?

Reply:
if it doesn't break anything then makes sense

Reply:
Do you mean I should add the `native-image.properties` file and initialize the package `io.micronaut.http.cookie` at build time or leave it as is and don't add any native image configuration?